bgpd/bgp_route.c

```diff
@@ -2482,7 +2482,11 @@ bool subgroup_announce_check(struct bgp_dest *dest, struct bgp_path_info *pi,
 	 */
 	if (NEXTHOP_IS_V6) {
 		attr->mp_nexthop_len = BGP_ATTR_NHLEN_IPV6_GLOBAL;
-		if ((CHECK_FLAG(peer->af_flags[afi][safi],
+		if (CHECK_FLAG(peer->af_flags[afi][safi],
+				PEER_FLAG_NEXTHOP_LOCAL_UNCHANGED)
+		     && !IN6_IS_ADDR_LINKLOCAL(&attr->mp_nexthop_local)) {
+			zlog_debug("No link-local next-hop. Ignoring other checks");
```

Comment:
We can't accept unguarded zlog_debug(), it must be printed only if debug is enabled. Even more, this is very naked logging, please use the format as others (including peer, prefix, ...).

Reply:
removed this change altogether